### PR TITLE
Add docs for cluster setup

### PR DESCRIPTION
**Pull request by gitanshu**

### CODE REVIEW

File: docs/user/cluster_setup.md [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+Write the following config options in a file `~/.prefect/config.toml`:
+
+```toml
+backend = "server"
+
+[server]
+host = "http://<Prefect server ip>"
+
+    [server.ui]
```
Comment (ryanjulian) [inline docs/user/cluster_setup.md:76]: do people usually indent toml documents?

Comment (gitanshu) [reply docs/user/cluster_setup.md:76]: I used prefect's main config as base and they were using indentation. My quick google search doesn't show any strong opinions about it.

**Review by nicolengsy — APPROVED** [review]

File: docs/user/cluster_setup.md [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+## More info on `dask-worker`s and resources
+
+### Resources
+Resources are completely arbitrary and abstract. That is, you could replace
+TOKEN with GPU or CPU or CHICKEN. A resource is just a labeled quantity which
+tasks can declare they require exclusively. A task's resource claims decrement
+available resources on the worker process, and the worker process will not
+schedule any task which would allow its resource counters to go negative.
+
+**IMPORTANT:** Resource declarations apply PER PROCESS regardless of invocation
```
Comment (maliesa96) [inline docs/user/cluster_setup.md:193]:
```suggestion
**IMPORTANT:** Resource declarations apply PER PROCESS regardless of invocation.
```

**Comment by maliesa96** [comment]
Nice!

**Review by maliesa96 — APPROVED** [review]

**Comment by gitanshu** [comment]
@mergifyio rebase

**Comment by gitanshu** [comment]
@mergifyio rebase

**Comment by gitanshu** [comment]
@Mergifyio rebase

**Comment by gitanshu** [comment]
@mergifyio rebase

**Review by krzentner — APPROVED** [review]

**Pull request merged**